src/porepy/numerics/ad/grid_operators.py

```diff
@@ -875,7 +960,7 @@ def parse(self, mdg: pp.MixedDimensionalGrid) -> np.ndarray:
             data = mdg.interface_data(intf)
             val.append(data[pp.PARAMETERS][self.param_keyword][self.array_keyword])
         if len(val) > 0:
-            return sps.diags(np.hstack([v for v in val]))
+            return sps.diags(np.hstack([v for v in val]), format="csr")
```

Comment:
It may not matter here, but I think `diag` matrices are considerably faster to operate with than `csr,` even if the csc represents a diagonal matrix.

Reply:
Do you mean a different format with the same constructur method or a different constructor?